### ISSUE TITLE
Double.NaN serializes but does not deserialize

### ISSUE THREAD

**Issue by ianohara** [json4s/json4s#246]
Here's a minimal demonstration of the issue.  My `json4s` version is `3.2.11`:

```
Welcome to Scala version 2.11.5 (OpenJDK 64-Bit Server VM, Java 1.7.0_79).
Type in expressions to have them evaluated.
Type :help for more information.

scala> import org.json4s._; import org.json4s.native.JsonMethods._; import org.json4s.Extraction; implicit val fmts = org.json4s.DefaultFormats
import org.json4s._
import org.json4s.native.JsonMethods._
import org.json4s.Extraction
fmts: org.json4s.DefaultFormats.type = org.json4s.DefaultFormats$@58eac8bb

scala> compact(render(Extraction.decompose(Double.NaN)))
res0: String = NaN

scala> parse(res0)
org.json4s.ParserUtil$ParseException: unknown token N
Near: N
  at org.json4s.native.JsonParser$Parser.fail(JsonParser.scala:207)
  at org.json4s.native.JsonParser$Parser.nextToken(JsonParser.scala:294)
  at org.json4s.native.JsonParser$$anonfun$1.apply(JsonParser.scala:159)
  at org.json4s.native.JsonParser$$anonfun$1.apply(JsonParser.scala:117)
  at org.json4s.native.JsonParser$.parse(JsonParser.scala:105)
  at org.json4s.native.JsonParser$.parse(JsonParser.scala:53)
  at org.json4s.native.JsonMethods$class.parse(JsonMethods.scala:11)
  at org.json4s.native.JsonMethods$.parse(JsonMethods.scala:62)
  ... 43 elided
```

I know `NaN` and `Infinity` are not allowed by the json spec [1] but since it is commonly allowed (or configurably allowed) by json libraries, and an exception was made for `Infinity` [2] I wonder if an exception can be made for `NaN` as well.

If there's already a way to globally configure `json4s` to accept `NaN` that I have missed, please let me know and sorry for the noise!

[1] - Page 3 in http://www.ecma-international.org/publications/files/ECMA-ST/ECMA-404.pdf
[2] - https://github.com/json4s/json4s/pull/159

**Comment by nornagon** [comment]
FWIW Chrome serializes both `NaN` and `Infinity` as `null` in JSON:

```
> JSON.stringify(NaN)
"null"
> JSON.stringify(Infinity)
"null"
```

**Comment by michaelvashchinsky** [comment]
+1

**Comment by seratch** [comment]
If there is no objection, null instead looks reasonable for me.

**Comment by ianohara** [comment]
Hey @nornagon and @seratch.  Consistency with Chrome is nice, but this doesn't fix the deserialization issue like #159 does for `Infinity`.

**Comment by nornagon** [comment]
There doesn't appear to be any way to build valid JSON that deserializes to
NaN.
